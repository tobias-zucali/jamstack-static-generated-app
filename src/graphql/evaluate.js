const { graphql } = require('graphql')

const { root, schema } = require('./schema.js')


module.exports = async ({
  query,
  variables,
  operationName,
}) => {
  const result = await graphql(
    schema,
    query,
    root,
    null,
    variables,
    operationName
  )

  return result
}
