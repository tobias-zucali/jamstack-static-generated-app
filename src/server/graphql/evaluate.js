import { graphql } from 'graphql'

import schema from './schema'


export default async function evaluate({
  query,
  variables,
  operationName,
}) {
  const result = await graphql(
    schema,
    query,
    null,
    null,
    variables,
    operationName
  )

  return result
}
