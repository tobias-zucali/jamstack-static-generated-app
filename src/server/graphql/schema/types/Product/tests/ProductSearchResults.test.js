import evaluate from 'server/graphql/evaluate'


describe('graphql/searchProducts', () => {
  it('returns all products with empty string', async () => {
    await expect(
      evaluate({
        query: `
            {
              searchProducts(searchString: "") {
                nodes {
                  node {
                    slug
                    name
                  }
                }
              }
            }
          `,
      })
    ).resolves.toMatchSnapshot()
  })

  it('returns all products containing "ap"', async () => {
    await expect(
      evaluate({
        query: `
            {
              searchProducts(searchString: "ap") {
                nodes {
                  matches {
                    product {
                      pre
                      match
                      post
                    }
                    manufacturer {
                      pre
                      match
                      post
                    }
                    category {
                      pre
                      match
                      post
                    }
                  }
                  node {
                    slug
                    name
                  }
                }
              }
            }
          `,
      })
    ).resolves.toMatchSnapshot()
  })

  it('returns all products with match in manufacturer', async () => {
    await expect(
      evaluate({
        query: `
            {
              searchProducts(searchString: "earth") {
                nodes {
                  matches {
                    product {
                      pre
                      match
                      post
                    }
                    manufacturer {
                      pre
                      match
                      post
                    }
                    category {
                      pre
                      match
                      post
                    }
                  }
                  node {
                    slug
                    name
                  }
                }
              }
            }
          `,
      })
    ).resolves.toMatchSnapshot()
  })

  it('returns all products with match in category', async () => {
    await expect(
      evaluate({
        query: `
            {
              searchProducts(searchString: "fruit") {
                nodes {
                  matches {
                    product {
                      pre
                      match
                      post
                    }
                    manufacturer {
                      pre
                      match
                      post
                    }
                    category {
                      pre
                      match
                      post
                    }
                  }
                  node {
                    slug
                    name
                  }
                }
              }
            }
          `,
      })
    ).resolves.toMatchSnapshot()
  })
})
