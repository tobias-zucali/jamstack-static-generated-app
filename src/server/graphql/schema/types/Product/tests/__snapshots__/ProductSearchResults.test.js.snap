// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql/searchProducts returns all products containing "ap" 1`] = `
Object {
  "data": Object {
    "searchProducts": Object {
      "nodes": Array [
        Object {
          "matches": Object {
            "category": null,
            "manufacturer": null,
            "product": Object {
              "match": "Ap",
              "post": "ple",
              "pre": "",
            },
          },
          "node": Object {
            "name": "Apple",
            "slug": "apple",
          },
        },
      ],
    },
  },
}
`;

exports[`graphql/searchProducts returns all products with empty string 1`] = `
Object {
  "data": Object {
    "searchProducts": Object {
      "nodes": Array [
        Object {
          "node": Object {
            "name": "Banana",
            "slug": "banana",
          },
        },
        Object {
          "node": Object {
            "name": "Manner Wafferl",
            "slug": "manner_wafferl",
          },
        },
        Object {
          "node": Object {
            "name": "Apple",
            "slug": "apple",
          },
        },
        Object {
          "node": Object {
            "name": "Gummi Bear",
            "slug": "gummi_bear",
          },
        },
        Object {
          "node": Object {
            "name": "Comfrey",
            "slug": "comfrey",
          },
        },
      ],
    },
  },
}
`;

exports[`graphql/searchProducts returns all products with match in category 1`] = `
Object {
  "data": Object {
    "searchProducts": Object {
      "nodes": Array [
        Object {
          "matches": Object {
            "category": Object {
              "match": "Fruit",
              "post": "",
              "pre": "",
            },
            "manufacturer": null,
            "product": null,
          },
          "node": Object {
            "name": "Banana",
            "slug": "banana",
          },
        },
        Object {
          "matches": Object {
            "category": Object {
              "match": "Fruit",
              "post": "",
              "pre": "",
            },
            "manufacturer": null,
            "product": null,
          },
          "node": Object {
            "name": "Apple",
            "slug": "apple",
          },
        },
      ],
    },
  },
}
`;

exports[`graphql/searchProducts returns all products with match in manufacturer 1`] = `
Object {
  "data": Object {
    "searchProducts": Object {
      "nodes": Array [
        Object {
          "matches": Object {
            "category": null,
            "manufacturer": Object {
              "match": "Earth",
              "post": " Inc.",
              "pre": "Mother ",
            },
            "product": null,
          },
          "node": Object {
            "name": "Banana",
            "slug": "banana",
          },
        },
        Object {
          "matches": Object {
            "category": null,
            "manufacturer": Object {
              "match": "Earth",
              "post": " Inc.",
              "pre": "Mother ",
            },
            "product": null,
          },
          "node": Object {
            "name": "Apple",
            "slug": "apple",
          },
        },
        Object {
          "matches": Object {
            "category": null,
            "manufacturer": Object {
              "match": "Earth",
              "post": " Inc.",
              "pre": "Mother ",
            },
            "product": null,
          },
          "node": Object {
            "name": "Comfrey",
            "slug": "comfrey",
          },
        },
      ],
    },
  },
}
`;
